detect_secrets/plugins/keyword.py

```diff
@@ -139,19 +139,24 @@ class KeywordDetector(BasePlugin):
 
     secret_type = 'Secret Keyword'
 
+    def __init__(self, keyword_exclude=None, **kwargs):
+        super(KeywordDetector, self).__init__(
+            keyword_exclude=keyword_exclude,
+        )
+
     def analyze_string_content(self, string, line_num, filename):
         output = {}
-
-        for identifier in self.secret_generator(
-            string,
-            filetype=determine_file_type(filename),
-        ):
-            secret = PotentialSecret(
-                self.secret_type,
-                filename,
-                identifier,
-                line_num,
-            )
+        if keyword_exclude and keyword_exclude != string:
```

Comment:
Before the `for identifier in self.secret_generator` loop we can do
```python
    if (
        self.keyword_exclude
        and self.keyword_exclude.search(string)
    ):
        return output
```
as I think string should always be a single-line.

Reply:
done